app/models/manageiq/providers/openstack/infra_manager.rb

```diff
@@ -73,4 +74,36 @@ def supports_authentication?(authtype)
   def self.event_monitor_class
     ManageIQ::Providers::Openstack::InfraManager::EventCatcher
   end
+
+  def clouds
+    overclouds = ManageIQ::Providers::Openstack::CloudManager.find(:all, :conditions => ["id != ? and provider_id = ?", id, provider_id])
```

Comment:
@rwsu yes we have provider.cloud_ems, btw. if it's missing, build it a relations like this, rather than method with sql query